rebasebot/bot.py

```diff
@@ -61,27 +62,30 @@ def _message_slack(webhook_url, msg):
 def _commit_go_mod_updates(gitwd, source: GitHubBranch):
     logging.info("Performing go modules update")
 
-    try:
-        # Reset go.mod and go.sum to make sure they are the same as in the source
-        for filename in ["go.mod", "go.sum"]:
-            if not os.path.exists(filename):
-                continue
-            gitwd.remotes.source.repo.git.checkout(f"source/{source.branch}", filename)
+    for filepath in glob.glob('./**/go.mod', recursive=True):
+        module_base_path = os.path.dirname(filepath)
```

Comment:
i wonder if we should check to see if a `vendor` folder exists in the same directory as the `go.mod`, as a backup check to ensure we want to run the update on that go.mod file?

Reply:
You mean before running `go mod vendor`, yes good point.
I think `go mod tidy` instead should always be ran, to check we are consistent with what imports we have in code.

Reply:
well yes, we should definitely check before running the `go mod vendor`. i was thinking we might not want to run it if we don't think the vendor dependencies are carried, but i think just running `go mod tidy` would be helpful to let us know if things are not in sync.

my main concern is with repos that have embedded tools which carry their own `go.mod`, but which we don't really care about from the rebasebot perspective. i'm probably thinking about it too much though, we can just check for the `vendor` folder before running the `go mod vendor` is probably fine.

Reply:
Agree 👍 

Added here: https://github.com/openshift-eng/rebasebot/pull/34/commits/7e45b6c7a4c1bec15e0a8d484f103f3d7574ac51

Reply:
One corner case came to my mind - if there was our commit where vendor folder was introduced and such commit was marked as <drop> and removed before this code engaged.

Reply:
I think in that case we would need to manually remove the folder when we entirely drop the "add vendor" commit (which I don't see happening frequently TBH).

Reply:
<drop> handler will just skip the commit. Think `vendor` folder it would never be created in this case.

> which I don't see happening frequently TBH

We didn't enable this IIRC :)

Reply:
Would be cool to have test for this, actually